### PR TITLE
Add support for the Blockstream Jade hww

### PR DESCRIPTION
**Pull request by JamieDriver**
Supports ssh and gpg, incl. ecdh/decryption.
Initially only supports curve 'nist256p1'.

### CODE REVIEW

File: libagent/device/jade.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    # Connect and fetch version info
+                    jade.connect()
+                    verinfo = jade.get_version_info()
+
+                    # Check minimum supported firmware version (ignore candidate/build parts)
+                    fwversion = semver.VersionInfo.parse(verinfo['JADE_VERSION'])
+                    if self.MIN_SUPPORTED_FW_VERSION > fwversion.finalize_version():
+                        msg = ('Outdated {} firmware for device. Please update using'
+                              ' a Blockstream Green companion app')
+                        raise ValueError (msg.format(fwversion))
```
Comment (romanz) [inline libagent/device/jade.py:63]:
```suggestion
                        raise ValueError(msg.format(fwversion))
```

File: agents/jade/setup.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+setup(
+    name='jade_agent',
+    version='0.1.0',
+    description='Using Blockstream Jade as hardware SSH agent',
+    author='Jamie C. Driver',
+    author_email='jamie@blockstream.com',
+    url='http://github.com/romanz/trezor-agent',
+    scripts=['jade_agent.py'],
+    install_requires=[
+        # FIXME: will need libagent version that includes jade support - 0.14.5 ??
```
Comment (romanz) [inline agents/jade/setup.py:13]: Let's fix it after this PR is merged :)

**Review by romanz — APPROVED** [review]
LGTM % a small comment

File: agents/jade/setup.py (hunk 2 above)
Comment (JamieDriver) [reply agents/jade/setup.py:13]: Sure.  Seems a little circular, so maybe it needs a couple of passes.  Whatever you deem best is good with me.

**Comment by JamieDriver** [comment]
Awesome, thanks.

**Comment by romanz** [comment]
Many thanks!

**Comment by romanz** [comment]
I fixed a few small issues here: https://github.com/romanz/trezor-agent/commit/df84c4c15f828e87d3e1a7ec681c5ddfa8a6f993
Please take a look :)

**Comment by JamieDriver** [comment]
Taken a look - yeah all good with me.
I re-installed from master, and tested ssh and gpg with my Jade hw and all seems good.
Many thanks for your time and indeed your swift action!  ;-)
Cheers.

**Pull request merged**